E-Tarik/coding-assignment
# Abdallah hassan/tech task

Review Notes:
* App.js is a little bit cluttered. need to seprate some functionality into custom hooks, components
* tests are not very well organized
For example  `movies starred and saved to watch later ` should be split into different scenarios
* some typos:  `remveAllWatchLater`
* some old JS syntax: string concatation instead of string template and usage of && instead of optional chaning

---------
This pull request
* address review feedback
* add grid display using css grid
* add react modal using create portal
* add inifinte scrolling in a custom hook using intersection observer
* fix and add testing
